src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts

```diff
@@ -34,86 +41,273 @@ declare var OpenSeadragon: any;
-        }
-      })
     }
+    this.viewer.clearOverlays();
+    this.viewer.open(tileSources);
+  }
 
-    private openImages() : void {
-      let imagesToDisplay = this.getImagesToDisplay();
-      let tileSources = [];
-      for (let i in imagesToDisplay) {
-        // construct IIIF tileSource configuration according to
-        // http://iiif.io/api/image/2.1/#technical-properties
-        // see also http://iiif.io/api/image/2.0/#a-implementation-notes
-        let sipiBasePath = imagesToDisplay[i].imageServerIIIFBaseURL + "/" + imagesToDisplay[i].imageFilename;
-        let width = imagesToDisplay[i].dimX;
-        let height = imagesToDisplay[i].dimY;
-        tileSources[i] = {
-          "tileSource": {
-           "@context": "http://iiif.io/api/image/2/context.json",
-           "@id": sipiBasePath,
-            "height": height,
-            "width": width,
-            "profile": [ "http://iiif.io/api/image/2/level2.json" ],
-            "protocol": "http://iiif.io/api/image",
-            "tiles": [{
-              "scaleFactors": [ 1, 2, 4, 8, 16, 32 ],
-              "width": 1024
-            }]
-          },
-          "x": i,
-          "y": 0
-        };
-      }
-      this.viewer.open(tileSources);
+  /**
+  * Returns an array containing all (non-preview) images in this.resources
+  * @returns {ReadStillImageFileValue[]}
+  */
+  private getImagesToDisplay() : ReadStillImageFileValue[] {
+    let fileValues : ReadStillImageFileValue[] = [];
+    for (let resource of this.resources) {
+      let fileValuesResource = resource.properties[AppConfig.hasStillImageFileValue];
+      Array.prototype.push.apply(fileValues, fileValuesResource);
     }
+    return fileValues.filter( (image) => { return !image.isPreview; } );
+  }
 
-    // If this.ignorePreviews == true:
-    //  returns a shallow copy of the this.fileValues array, omitting all entries with entry.isPreview == true.
-    // else:
-    //  returns the this.fileValues array itself
-    private getImagesToDisplay() : ReadStillImageFileValue[] {
-      if (this.ignorePreviews) {
-        return this.fileValues.filter( (image) => { return !image.isPreview; } );
-      } else {
-        return this.fileValues;
+  /**
+  * Returns an array containing all regions of all (non-preview) images in this.resources
+  * The returneed array has the form [ [ image0_region0, image0_region1, ... ], [ image1_region0, image1_region1 ], ... ]
+  * @returns {ReadResource[][]}
+  */
+  private getRegionsToDisplay() : ReadResource[][] {
+    let regions = [];
+    for (let resource of this.resources) {
+      let incomingRegions = resource.incoming.filter((resource) => { return resource.type == AppConfig.Region});
+      regions.push(incomingRegions);
+    }
+    return regions;
+  }
+
+  /**
+  * Adds a ROI-overlay to the viewer for every region of every (non-preview) image in this.resources
+  */
+  private renderRegions() : void {
```

Comment:
This component should and does not know, how parent components make use of it. 
IF parent components add incoming regions to resources in this.resources after this.resources has been assigned, they need to call updateRegions() like you described.
However, parent components might also reassign this.resources with another array of resources which already have incoming regions, or this.resources might contain incoming resources from the beginning. In this case, parent components do not need to make another call to updateRegions since ngOnChanges will fire and handle rendering of regions automatically. This is when case 1 applies.